File: EstructuraDeDatos/ClasesBase/ItemBase.cpp
// ItemBase.cpp
#include "ItemBase.h"
Elemento::Elemento()
{
 //ctor
}
Elemento::~Elemento()
{
 //dtor
}
int Elemento::Leer(int ind)
{
 cout << "ITEM[" << ind << "], clave (salir <= 0): ";
 cin >> m_Clave;
 if (m_Clave <= 0) return 0;
 if (m_Indicador > 1) {
 getchar(); // Limpiar el buffer
 cout << "dato (salir = NULO): ";
 gets(m_Dato);
 if (strlen(m_Dato) == 0) return 0;
 }
 m_Counter = 0;
 return 1;
}
int Elemento::Leer(FILE *F)
{
 int est = fscanf(F, "%d", &m_Clave);
 if (est < 0 || m_Clave <= 0) return 0;
 if (m_Indicador > 1) {
 fgetc(F); // Limpiar el buffer
 if (!fgets(m_Dato, 50, F)) return 0;
 m_Dato[strlen(m_Dato) - 1] = '\0';
 if (strlen(m_Dato) == 0) return 0;
 }
 m_Counter = 0;
 return 1;
}
void Elemento::Mostrar()
{
 cout << m_Clave << " ";
 if (m_Indicador == 1 || m_Indicador == 3)
 cout << m_Counter << " ";
 if (m_Indicador > 1)
 cout << m_Dato << endl;
}
void Elemento::Mostrar(FILE *G)
{
 fprintf(G, "%d ", m_Clave);
 if (m_Indicador == 1 || m_Indicador == 3)
 fprintf(G, "%d ", m_Counter);
 if (m_Indicador > 1)
 fprintf(G, "%s\n", m_Dato);
}
int Elemento::Compara(Elemento x)
{
 return m_Clave - x.m_Clave;
}
void Elemento::Copiar(Elemento x)
{
 *this = x;
}

File: EstructuraDeDatos/Python/ModulosPY/ItemBase.py
# ItemBase.py
class ItemBase:
    Indicador = 0
    def __init__(elm, clv, cuenta, dato):
        elm.clave = clv
        elm.cuenta = cuenta
        elm.dato = dato

    def Leer(elm, ind):
        val = int(input("ITEM[" + str(ind) + "], clave (salir <= 0): "))
        if val <= 0:
            return elm, 0
        elm.clave = val
        if ItemBase.Indicador == 0:
            return elm, 1
        cad = input("dato (salir = NULO): ")
        if len(cad) == 0:
            return elm, 0
        elm.dato = cad
        return elm, 1
    def Mostrar(x, cad):
        cad += str(x.clave)+ " "
        if ItemBase.Indicador == 0:
            return cad
        if ItemBase.Indicador >= 1:
            cad += str(x.cuenta)+ " "
        if ItemBase.Indicador == 1:
            return cad
        cad += str(x.dato)+ "\n "
        return cad
    def Comparar(elm, y):
        return elm.clave - y.clave